enhancements/metering/slim-install.md

```diff
@@ -0,0 +1,281 @@
+
+Note: Postgresql has no native support for creating a database if it doesn't already exist, so the following call will need to be made instead:
+
+```SQL
+SELECT 'CREATE DATABASE metering' WHERE NOT EXISTS (SELECT FROM pg_database WHERE datname = 'metering')
+```
+
+#### Outlining a Potential Migration Path
+
+There's currently an undocumented PUSH API for the reporting-operator to statically inject Prometheus data that matches a specific JSON format. This API is largely used in Metering end-to-end testing scenarios and is likely, not robust or stable enough to rely on for a full data set migration from existing Metering installations. In theory, a migration effort is technically possible, but a more robust API implementation would be needed.
+
+Also worth noting that making changes to this API would likely require backporting those changes to previously released versions of Metering.
+
+### Risks and Mitigations
+
+#### TLS and Authentication
+
+**Risk**: Need to support TLS and authentication for communication between the reporting-operator and the Postgres database instance.
+
+**Mitigation**: Postgres has native support for mutual SSL/TLS and client authentication. The reporting-operator, which is deployed with an Openshift auth-proxy sidecar container, can be configured to trust the Postgres instance. The metering-ansible-operator will be responsible for generating these client/server/root certificates for the components.
+
+## Design Details
+
+### Test Plan
+
+- Provide a mechanism for statically injecting cluster metric data into `ReportDataSource` database tables to ensure that the refactored list of `ReportQuery` queries match the expected `Report` output results. Those output results should also match what we expect in the non-slim-install version of Metering.
+- Ensure that the refactored list of `ReportDataSource` resources result in at least one row of metric data being dynamically imported from Prometheus into the corresponding table in Postges.
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+Only tech preview has been planned for this proposal.
+
+#### Tech Preview -> GA
+
+- Provide and document a data migration path for existing Metering installations to this new slimmed down architecture.
+- The Metering API versioning will be bumped from `v2alpha1` to `v2` for the CRDs.
+- Removal of support for the v1 Metering bundle.
+- Sufficient time for feedback.
+- Sufficient testing in the following areas: upgrade, downgrade, scale and e2e.
+
+### Upgrade / Downgrade Strategy
+
+Metering will continue to be packaged as an OLM-based Operator, so OLM will be responsible for orchestrating upgrades and downgrades between v2 Metering versions.
+
+The current state of this proposal upgrades between the v1 and v2 versions of the Operator won't be supported, and instead a new installation of the v2 Metering Operator will be required.
```

Comment:
can we get a little more detail on what the user experience is for the following scenarios (maybe even write these up as user scenarios/use cases?)

1) i'm an existing metering user, i upgrade to the new version of metering that supports slim install but i'm not ready to switch to it

2) i'm an existing metering user, i upgrade to the new version of metering that supports slim install and i want to switch to using the slim install

3) i'm a new user who wants to install metering and use the slim install

4) i'm a new user who wants to install metering and does not want to use the slim install